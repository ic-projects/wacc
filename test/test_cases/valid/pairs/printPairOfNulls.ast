-- Test: printPairOfNulls.wacc

-- Uploaded file: 
---------------------------------------------------------------
# print a pair of null pairs 

# Output:
# #addrs# = ((nil),(nil))

# Program:

begin
  pair(pair, pair) p = newpair(null, null) ;
  print p ;
  print " = (" ;
  pair(pair, pair) q = fst p ;
  print q ;
  print "," ;
  pair(int, bool) r = snd p ;
  print r ; 
  println ")"
end
---------------------------------------------------------------

-- Compiler Output:
-- Compiling...
-- Printing AST...
printPairOfNulls.ast contents are:
===========================================================
0	Program
1	  - int main()
2	    - DECLARE
3	      - TYPE
4	        - pair(basePair, basePair)
5	      - LHS
6	        - p
7	      - RHS
8	        - NEW_PAIR
9	          - FST
10	            - null
11	          - SND
12	            - null
13	    - PRINT
14	      - p
15	    - PRINT
16	      - " = ("
17	    - DECLARE
18	      - TYPE
19	        - pair(basePair, basePair)
20	      - LHS
21	        - q
22	      - RHS
23	        - FST
24	          - p
25	    - PRINT
26	      - q
27	    - PRINT
28	      - ","
29	    - DECLARE
30	      - TYPE
31	        - pair(int, bool)
32	      - LHS
33	        - r
34	      - RHS
35	        - SND
36	          - p
37	    - PRINT
38	      - r
39	    - PRINTLN
40	      - ")"
===========================================================
-- Finished

