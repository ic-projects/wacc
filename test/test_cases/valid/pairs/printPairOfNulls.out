-- Compiling...
-- Assembling and Linking...
-- Executing...
===========================================================
0x22008 = ((nil),(nil))
===========================================================
The exit code is 0.
-- Finished
